Spring Boot 2.0 apps fail on startup due to log4j 2.10.0 conflict with Jetty 9.4.5

```
org.eclipse.jetty.webapp.WebAppContext: Failed startup of context o.e.j.w.WebAppContext@4678c730{/,file:///var/lib/jetty/webapps/root/,UNAVAILABLE}{/root.war}
MultiException[java.lang.RuntimeException: Error scanning entry META-INF/versions/9/org/apache/logging/log4j/util/ProcessIdUtil.class from jar file:///var/lib/jetty/webapps/root/WEB-INF/lib/log4j-api-2.10.0.jar, java.lang.RuntimeException: Error scanning entry META-INF/versions/9/org/apache/logging/log4j/util/StackLocator.class from jar file:///var/lib/jetty/webapps/root/WEB-INF/lib/log4j-api-2.10.0.jar, java.lang.RuntimeException: Error scanning entry module-info.class from jar file:///var/lib/jetty/webapps/root/WEB-INF/lib/log4j-api-2.10.0.jar]
        at org.eclipse.jetty.annotations.AnnotationConfiguration.scanForAnnotations(AnnotationConfiguration.java:451)
        at org.eclipse.jetty.annotations.AnnotationConfiguration.configure(AnnotationConfiguration.java:364)
        at org.eclipse.jetty.webapp.WebAppContext.configure(WebAppContext.java:517)
        at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1458)
        at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:785)
        at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:261)
        at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:545)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
        at org.eclipse.jetty.deploy.bindings.StandardStarter.processBinding(StandardStarter.java:41)
        at org.eclipse.jetty.deploy.AppLifeCycle.runBindings(AppLifeCycle.java:188)
        at org.eclipse.jetty.deploy.DeploymentManager.requestAppGoal(DeploymentManager.java:502)
        at org.eclipse.jetty.deploy.DeploymentManager.addApp(DeploymentManager.java:150)
        at org.eclipse.jetty.deploy.providers.ScanningAppProvider.fileAdded(ScanningAppProvider.java:180)
        at org.eclipse.jetty.deploy.providers.WebAppProvider.fileAdded(WebAppProvider.java:447)
        at org.eclipse.jetty.deploy.providers.ScanningAppProvider$1.fileAdded(ScanningAppProvider.java:64)
        at org.eclipse.jetty.util.Scanner.reportAddition(Scanner.java:610)
        at org.eclipse.jetty.util.Scanner.reportDifferences(Scanner.java:529)
        at org.eclipse.jetty.util.Scanner.scan(Scanner.java:392)
        at org.eclipse.jetty.util.Scanner.doStart(Scanner.java:313)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
        at org.eclipse.jetty.deploy.providers.ScanningAppProvider.doStart(ScanningAppProvider.java:150)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
        at org.eclipse.jetty.deploy.DeploymentManager.startAppProvider(DeploymentManager.java:564)
        at org.eclipse.jetty.deploy.DeploymentManager.doStart(DeploymentManager.java:239)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:131)
        at org.eclipse.jetty.server.Server.start(Server.java:452)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:113)
        at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:113)
        at org.eclipse.jetty.server.Server.doStart(Server.java:419)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
        at org.eclipse.jetty.xml.XmlConfiguration$1.run(XmlConfiguration.java:1511)
        at org.eclipse.jetty.xml.XmlConfiguration$1.run(XmlConfiguration.java:1438)
        at java.security.AccessController.doPrivileged(Native Method)
        at org.eclipse.jetty.xml.XmlConfiguration.main(XmlConfiguration.java:1437)
Caused by: MultiException[java.lang.RuntimeException: Error scanning entry META-INF/versions/9/org/apache/logging/log4j/util/ProcessIdUtil.class from jar file:///var/lib/jetty/webapps/root/WEB-INF/lib/log4j-api-2.10.0.jar, java.lang.RuntimeException: Error scanning entry META-INF/versions/9/org/apache/logging/log4j/util/StackLocator.class from jar file:///var/lib/jetty/webapps/root/WEB-INF/lib/log4j-api-2.10.0.jar, java.lang.RuntimeException: Error scanning entry module-info.class from jar file:///var/lib/jetty/webapps/root/WEB-INF/lib/log4j-api-2.10.0.jar]
        at org.eclipse.jetty.annotations.AnnotationParser.parseJar(AnnotationParser.java:878)
        at org.eclipse.jetty.annotations.AnnotationParser.parse(AnnotationParser.java:837)
        at org.eclipse.jetty.annotations.AnnotationConfiguration$ParserTask.call(AnnotationConfiguration.java:160)
        at org.eclipse.jetty.annotations.AnnotationConfiguration$1.run(AnnotationConfiguration.java:463)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:672)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:590)
        at java.lang.Thread.run(Thread.java:748)
        Suppressed: java.lang.RuntimeException: Error scanning entry META-INF/versions/9/org/apache/logging/log4j/util/StackLocator.class from jar file:///var/lib/jetty/webapps/root/WEB-INF/lib/log4j-api-2.10.0.jar
                at org.eclipse.jetty.annotations.AnnotationParser.parseJar(AnnotationParser.java:891)
                ... 6 more
        Caused by: java.lang.IllegalArgumentException
                at org.objectweb.asm.ClassReader.<init>(Unknown Source)
                at org.objectweb.asm.ClassReader.<init>(Unknown Source)
                at org.objectweb.asm.ClassReader.<init>(Unknown Source)
                at org.eclipse.jetty.annotations.AnnotationParser.scanClass(AnnotationParser.java:959)
                at org.eclipse.jetty.annotations.AnnotationParser.parseJarEntry(AnnotationParser.java:940)
                at org.eclipse.jetty.annotations.AnnotationParser.parseJar(AnnotationParser.java:887)
                ... 6 more
        Suppressed: java.lang.RuntimeException: Error scanning entry module-info.class from jar file:///var/lib/jetty/webapps/root/WEB-INF/lib/log4j-api-2.10.0.jar
                at org.eclipse.jetty.annotations.AnnotationParser.parseJar(AnnotationParser.java:891)
                ... 6 more
        Caused by: java.lang.IllegalArgumentException
                at org.objectweb.asm.ClassReader.<init>(Unknown Source)
                at org.objectweb.asm.ClassReader.<init>(Unknown Source)
                at org.objectweb.asm.ClassReader.<init>(Unknown Source)
                at org.eclipse.jetty.annotations.AnnotationParser.scanClass(AnnotationParser.java:959)
                at org.eclipse.jetty.annotations.AnnotationParser.parseJarEntry(AnnotationParser.java:940)
                at org.eclipse.jetty.annotations.AnnotationParser.parseJar(AnnotationParser.java:887)
                ... 6 more
Caused by: java.lang.RuntimeException: Error scanning entry META-INF/versions/9/org/apache/logging/log4j/util/ProcessIdUtil.class from jar file:///var/lib/jetty/webapps/root/WEB-INF/lib/log4j-api-2.10.0.jar
        at org.eclipse.jetty.annotations.AnnotationParser.parseJar(AnnotationParser.java:891)
        ... 6 more
Caused by: java.lang.IllegalArgumentException
        at org.objectweb.asm.ClassReader.<init>(Unknown Source)
        at org.objectweb.asm.ClassReader.<init>(Unknown Source)
        at org.objectweb.asm.ClassReader.<init>(Unknown Source)
        at org.eclipse.jetty.annotations.AnnotationParser.scanClass(AnnotationParser.java:959)
        at org.eclipse.jetty.annotations.AnnotationParser.parseJarEntry(AnnotationParser.java:940)
        at org.eclipse.jetty.annotations.AnnotationParser.parseJar(AnnotationParser.java:887)
        ... 6 more

```

See: https://github.com/eclipse/jetty.project/issues/1797

It looks like Jetty 9.4.7 addressed the issue. So, we should just upgrade to it.

@gregw FYI

--- a/pom.xml
+++ b/pom.xml
@@ -31,8 +31,8 @@
     <maven.build.timestamp.format>yyyy-MM-dd-HH-mm</maven.build.timestamp.format>
     <gcloud.api.bom.version>0.32.0-alpha</gcloud.api.bom.version>
     <jetty9.minor.version>4</jetty9.minor.version>
-    <jetty9.dot.version>5</jetty9.dot.version>
-    <jetty9.version>9.${jetty9.minor.version}.${jetty9.dot.version}.v20170502</jetty9.version>
+    <jetty9.dot.version>8</jetty9.dot.version>
+    <jetty9.version>9.${jetty9.minor.version}.${jetty9.dot.version}.v20171121</jetty9.version>
     <docker.tag.prefix></docker.tag.prefix>
     <docker.tag.short>${docker.tag.prefix}9.${jetty9.minor.version}</docker.tag.short>
     <docker.tag.long>${docker.tag.prefix}9.${jetty9.minor.version}-${maven.build.timestamp}</docker.tag.long>

